### PR TITLE
exodus-lambda crashed when missing content-type in DynamoDB [RHELDST-…

### PR DESCRIPTION
**Pull request by dichen16**
…8905]

When a client requests an item with no content_type set, the response
should include a Content-Type of "application/octet-stream".

Add a new test case cover the empty content_type scenario.

### CODE REVIEW

File: support/reftest/reftest [hunk 1]
```diff
@@ -38,7 +38,9 @@ class DBHandler:
     ):
         if not from_date:
             from_date = self.default_from_date
-        return self.dynamodb.put_item(
+        if not content_type:
+            content_type = ""
```
Comment (rohanpm) [inline support/reftest/reftest:42]: If the intent here was to cover what happens when there's no content_type, I don't think this quite achieves that, does it?  This will store content_type of `""` (empty string) on the created items.

Most items in the DB right now have *no* content_type at all, which is slightly different.  Should you do the same here?

File: tests/functions/test_origin_request.py [hunk 2]
```diff
@@ -74,6 +79,7 @@
         "no alias keywords",
         "releasever alias",
         "layered rhui, releasever alias",
+        "no content_type"
```
Comment (rohanpm) [inline tests/functions/test_origin_request.py:82]: Similarly as the comment against reftest, this does not cover the realistic case properly.  Realistically, items can have no content_type at all which is different from having a content_type of an empty string.  In fact the old code would already not crash if an item had content_type of `""`, so adding a test for that value is not going to prove that the crash was fixed.

I think you ought to edit the test (or add a new one) which truly tests the behavior on *no* content_type, rather than a content_type of `""`.

**Review by rohanpm — CHANGES_REQUESTED** [review]

File: support/reftest/reftest [hunk 3]
```diff
@@ -38,15 +38,24 @@ class DBHandler:
     ):
         if not from_date:
             from_date = self.default_from_date
-        return self.dynamodb.put_item(
-            TableName=self.table,
-            Item={
+        if not content_type:
+            item = {
```
Comment (rohanpm) [inline support/reftest/reftest:42]: Seems like it'd be better to not duplicate the setting of most of the item fields, since that invites bugs where someone updates one path and forgets the other.

Would it make more sense to write it like this:

```python
    item = {
       "web_uri": # ...,
       # ...rest of fields except content_type
    }
    if content_type:
        item["content_type"] = {"S": content_type}
```

So that there are not two places where web_uri, object_key etc are being set.

**Review by rohanpm — APPROVED** [review]

File: tests/functions/test_origin_request.py [hunk 4]
```diff
@@ -221,3 +221,60 @@ def test_origin_request_definitions_cache(
...
+        ),
+    ],
+    ids=[
+        "no content_type",
+    ],
+)
+@mock.patch("boto3.client")
+@mock.patch("exodus_lambda.functions.origin_request.cachetools")
+@mock.patch("exodus_lambda.functions.origin_request.datetime")
+def test_origin_request_no_content_type(
```
Comment (nathanegillett) [inline tests/functions/test_origin_request.py:241]: Minor:
Setup for this test seems odd to me. It's using parametrize but only has one case and in that case both parameters are identical. Maybe it's meant to be expanded later? Though, we probably should only build things around what's strictly necessary rather than future potential.

**Review by nathanegillett — APPROVED** [review]

**Review by crungehottman — APPROVED** [review]
